Flaky test PodTest.testExecEphemeralContainer

```
Error:  Tests run: 32, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.80 s <<< FAILURE! -- in io.fabric8.kubernetes.client.mock.PodTest
Error:  io.fabric8.kubernetes.client.mock.PodTest.testExecEphemeralContainer -- Time elapsed: 10.06 s <<< ERROR!
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:109)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:102)
	at io.fabric8.kubernetes.client.utils.Utils.waitUntilReadyOrFail(Utils.java:187)
	at io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl.setupConnectionToPod(PodOperationsImpl.java:387)
	at io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl.exec(PodOperationsImpl.java:293)
	at io.fabric8.kubernetes.client.mock.PodTest.testExecEphemeralContainer(PodTest.java:1048)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
Caused by: java.util.concurrent.TimeoutException: not ready after 10000 MILLISECONDS
	... 7 more
```

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java
@@ -374,17 +374,30 @@ public class PodOperationsImpl extends HasMetadataOperation<Pod, PodList, PodRes
   private ExecWebSocketListener setupConnectionToPod(URI uri) {
     ExecWebSocketListener execWebSocketListener = new ExecWebSocketListener(getContext(), this.context.getExecutor(),
         this.getKubernetesSerialization());
+    long requestTimeoutMs = getRequestConfig().getRequestTimeout();
     CompletableFuture<WebSocket> startedFuture = httpClient.newWebSocketBuilder()
         .subprotocol("v4.channel.k8s.io")
         .uri(uri)
-        .connectTimeout(getRequestConfig().getRequestTimeout(), TimeUnit.MILLISECONDS)
+        .connectTimeout(requestTimeoutMs, TimeUnit.MILLISECONDS)
         .buildAsync(execWebSocketListener);
     startedFuture.whenComplete((w, t) -> {
       if (t != null) {
         execWebSocketListener.onError(w, t);
       }
     });
-    Utils.waitUntilReadyOrFail(startedFuture, getRequestConfig().getRequestTimeout(), TimeUnit.MILLISECONDS);
+    final long startNanos = System.nanoTime();
+    try {
+      Utils.waitUntilReadyOrFail(startedFuture, requestTimeoutMs, TimeUnit.MILLISECONDS);
+    } catch (RuntimeException e) {
+      // Diagnostic for #7737: capture state when the WS-upgrade future fails to complete.
+      LOG.error("Exec WebSocket upgrade did not complete after {}ms (timeout {}ms). uri={} futureDone={} thread={}",
+          (System.nanoTime() - startNanos) / 1_000_000L,
+          requestTimeoutMs,
+          uri,
+          startedFuture.isDone(),
+          Thread.currentThread().getName());
+      throw e;
+    }
     return execWebSocketListener;
   }
 

